File: 171.excel-sheet-column-number.py
#
# @lc app=leetcode id=171 lang=python3
#
# [171] Excel Sheet Column Number
#

# Time:  O(n)
# Space: O(1)

# @lc code=start
class Solution:
    def titleToNumber(self, s: str) -> int:
        res = 0
        for i in range(len(s)):
            res = res * 26 + ord(s[i]) - 64
        return res


# @lc code=end

File: 171.excel-sheet-column-number.go
/*
 * @lc app=leetcode id=171 lang=golang
 *
 * [171] Excel Sheet Column Number
 */

package leetcode

// @lc code=start
func titleToNumber(columnTitle string) int {
	res := 0
	for i := 0; i < len(columnTitle); i++ {
		res = res*26 + int(columnTitle[i]) - 64
	}
	return res
}

// @lc code=end
